workflow/common/convert.go

```diff
@@ -7,26 +7,33 @@ import (
 	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
 )
 
-func ConvertCronWorkflowToWorkflow(cronWf *wfv1.CronWorkflow) (*wfv1.Workflow, error) {
-	newTypeMeta := metav1.TypeMeta{
-		Kind:       workflow.WorkflowKind,
-		APIVersion: cronWf.TypeMeta.APIVersion,
-	}
-
-	newObjectMeta := metav1.ObjectMeta{}
-	newObjectMeta.GenerateName = cronWf.Name + "-"
+func ConvertCronWorkflowToWorkflow(cronWf *wfv1.CronWorkflow) *wfv1.Workflow {
+	wf := toWorkflow(cronWf.TypeMeta, cronWf.ObjectMeta, cronWf.Spec.WorkflowSpec)
+	wf.ObjectMeta.GetLabels()[LabelKeyCronWorkflow] = cronWf.Name
+	wf.SetOwnerReferences(append(wf.GetOwnerReferences(), *metav1.NewControllerRef(cronWf, wfv1.SchemeGroupVersion.WithKind(workflow.CronWorkflowKind))))
+	return wf
+}
 
-	newObjectMeta.Labels = make(map[string]string)
-	newObjectMeta.Labels[LabelKeyCronWorkflow] = cronWf.Name
-	if instanceId, ok := cronWf.GetLabels()[LabelKeyControllerInstanceID]; ok {
-		newObjectMeta.Labels[LabelKeyControllerInstanceID] = instanceId
-	}
+func ConvertWorkflowTemplateToWorkflow(template *wfv1.WorkflowTemplate) *wfv1.Workflow {
+	wf := toWorkflow(template.TypeMeta, template.ObjectMeta, template.Spec.WorkflowSpec)
```

Comment:
do we want to add a label to indicate the template use?

Reply:
@jessesuen when you create a workflow from a template, do you think it should have the same owner relationship a cron has to its workflows?

Reply:
yes. I am pushing the change

Reply:
> @jessesuen when you create a workflow from a template, do you think it should have the same owner relationship a cron has to its workflows?

I think so too, although this will delete all `Workflows` from a `WorkflowTemplate` if the `WorkflowTempalte` is deleted.

Reply:
we are just adding the only label. if we add ownerreference then  workflow will get deleted